### DISCUSSION TITLE
Cancel Trino Query Execution API

### DISCUSSION THREAD

**Discussion by arunkumaralagesan** [trinodb/trino-python-client#363] [Q&A]
Hi Team, we at Linkedin are using `Sqlalchemy@1.3.24` to query trino cluster and and would like to know, if there is any API to cancel a running query, so that the trino queue gets cleared up for further queries instead of closing the connection?
Went through documentation and other sources. No luck yet.
Thanks in advance!

**Comment by hashhar** [comment]
This is more a SQLAlchemy question. From this client's perspective you can close a connection to cancel the query or call the `cancel` method on the `Cursor`.

I don't think SQLAlchemy offers any methods to do that.

You can try to use the `system.runtime.kill_query` procedure https://trino.io/docs/current/connector/system.html#system-connector-procedures
